share/picongpu/dockerfiles/ubuntu-1604/Dockerfile

```diff
@@ -1,80 +1,72 @@
-FROM       nvidia/cuda:8.0-devel
+FROM       nvidia/cuda:9.0-base
 MAINTAINER Axel Huebl <a.huebl@hzdr.de>
 
 # docker and image environment
 ENV        DEBIAN_FRONTEND=noninteractive \
-           PICONGPU_BRANCH="0.3.0" \
-           HOME=/home/picongpu \
-           PATH=/home/picongpu/src/spack/bin:$PATH
+           FORCE_UNSAFE_CONFIGURE=1 \
+           SPACK_ROOT=/usr/local \
+           SPACK_EXTRA_REPO=/usr/local/share/spack-repo \
+           PIC_PACKAGE='picongpu@develop+isaac backend=cuda'
 
 # install minimal spack dependencies
 #   - adds gfortran for spack's openmpi package
-#   - adds the standard editors for users: vim, nano, emacs
+#   - adds the standard editors for users: vim, nano
+#     (excluded: emacs-nox pulls up to 90MB in)
 #   - adds build tools which are NOT linked in the final app
+#   - selected cuda toolkit components
+# need more cuda toolkit components?
+# check:   apt-get update && apt-cache search cuda && apt-cache search nvcc
 RUN        apt-get update && \
            apt-get install -y --no-install-recommends \
               autoconf \
               build-essential \
               ca-certificates \
               coreutils \
+              cuda-command-line-tools-$CUDA_PKG_VERSION \
+              cuda-core-$CUDA_PKG_VERSION \
+              cuda-cudart-dev-$CUDA_PKG_VERSION \
+              cuda-curand-dev-$CUDA_PKG_VERSION \
+              cuda-minimal-build-$CUDA_PKG_VERSION \
+              cuda-misc-headers-$CUDA_PKG_VERSION \
+              cuda-nvml-dev-$CUDA_PKG_VERSION \
               curl \
-              emacs \
```

Comment:
Nooo ß emacs is gone. 😱 
But seriously, there are still nano and vim. Should we just support one editor?

Reply:
Yes I know, I would have liked to keep it.

I added a comment above. `emacs-nox` is pulling in about 90MB to the image which is pretty heavy for a docker image. (the base ubuntu is only 127 MB).

`nano` and/or `vim` did not really weight that much, so I kept both.